Add boolean operators to Mathematical Operators (#35996)
JuliaLang/julia: doc/src/manual/mathematical-operations.md

This is useful for preventing the propagation of `NaN` values in quantities that are known to be zero. See [Knuth (1992)](https://arxiv.org/abs/math/9205211) for motivation.

## Boolean Operators

The following [boolean operators](https://en.wikipedia.org/wiki/Boolean_algebra#Operations) are supported on Bool:

| Expression | Name                                  |
|:---------- |:--------------------------------------|
| `!x`       | not                                   |
| `x && y`   | [short-circuiting and][short-circuit] |
| `x \|\| y` | [short-circuiting or][short-circuit]  |

Note that Bool is an integer type and all the usual promotion rules and numeric operators are also defined on it.

> 

Author: ```suggestion

[short-circuit]: https://docs.julialang.org/en/v1/manual/control-flow/#Short-Circuit-Evaluation-1
```

Somehow lost this link

This is useful for preventing the propagation of `NaN` values in quantities that are known to be zero. See [Knuth (1992)](https://arxiv.org/abs/math/9205211) for motivation.

## Boolean Operators

> The following [boolean operators](https://en.wikipedia.org/wiki/Boolean_algebra#Operations) are supported on Bool:

Reviewer: ```suggestion
The following [Boolean operators](https://en.wikipedia.org/wiki/Boolean_algebra#Operations) are supported on `Bool`:
```

Author: Thanks :)

| Expression | Name                                  |
|:---------- |:--------------------------------------|
| `!x`       | not                                   |
| `x && y`   | [short-circuiting and][short-circuit] |
| `x \|\| y` | [short-circuiting or][short-circuit]  |

> | `x == y`   | equivalence                           |

Author: ```suggestion
```

> Note that Bool is an integer type and all the usual promotion rules and numeric operators are also defined on it.

Reviewer: ```suggestion
Note that `Bool` is an integer type and all the usual promotion rules and numeric operators are also defined on it.
```

This is useful for preventing the propagation of `NaN` values in quantities that are known to be zero. See [Knuth (1992)](https://arxiv.org/abs/math/9205211) for motivation.

## Boolean Operators

The following [boolean operators](https://en.wikipedia.org/wiki/Boolean_algebra#Operations) are supported on Bool:

| Expression | Name                                  |
|:---------- |:--------------------------------------|
| `!x`       | not                                   |

> | `x & y`    | and                                   |
> | `x \| y`   | or                                    |
> | `x ⊻ y`    | xor (exclusive or)                    |

Reviewer 2: ```suggestion
```

These are documented below and I'd rather push folks towards using `&&` and `||` for boolean logic as it has the precedence they likely want.

Author: Sure. Done.

| `x && y`   | [short-circuiting and][short-circuit] |
| `x \|\| y` | [short-circuiting or][short-circuit]  |
| `x == y`   | equivalence                           |

> Note that Bool is an integer type and all the usual promotions rules and numeric operators are also defined on it (in fact, `&`, `|`, and `⊻` *are* bitwise operators).

Author: ```suggestion
Note that Bool is an integer type and all the usual promotion rules and numeric operators are also defined on it.
```
